corelib-storage/src/main/java/eu/europeana/corelib/solr/derived/AttributionConverter.java

```diff
@@ -33,27 +34,29 @@ private void processItemURI(Attribution attribution, WebResourceImpl wRes) {
     }
 
     private void processTCD(Attribution attribution, WebResourceImpl wRes) {
+        //map to store the creator values and process it
+        Map<String, List<String>> creatorMap = new HashMap<>();
         // get the creator first from web resource level
-        if (attribution.getCreator().size() == 0 && isNotBlank(wRes.getDcCreator())) {
-            attribution.getCreator().putAll(concatLangawareMap(wRes.getDcCreator()));
+        if (creatorMap.size() == 0 && isNotBlank(wRes.getDcCreator())) {
+            creatorMap.putAll(createCreatorMap(wRes.getDcCreator()));
         }
         //if still empty get, title, creator and date from proxy
-        checkProxy(((AggregationImpl) wRes.getParentAggregation()).getParentBean().getProxies(), attribution);
+        checkProxy(((AggregationImpl) wRes.getParentAggregation()).getParentBean().getProxies(), attribution, creatorMap);
         // if there was no title found in the proxy, get it from the record object itself
         if (attribution.getTitle().size() == 0 && ArrayUtils.isNotEmpty(((AggregationImpl) wRes.getParentAggregation()).getParentBean().getTitle())) {
             attribution.getTitle().put("", collectListInLines(Arrays.asList(stripEmptyStrings(((AggregationImpl) wRes.getParentAggregation()).getParentBean().getTitle()))));
         }
         //check europeana aggregation if TCD is still empty
-        checkEuropeanaAggregration(attribution, ((AggregationImpl) wRes.getParentAggregation()).getParentBean().getEuropeanaAggregation());
-
-        checkCreatorLabel(attribution, (List<Agent>) ((AggregationImpl) wRes.getParentAggregation()).getParentBean().getAgents());
+        checkEuropeanaAggregration(attribution, ((AggregationImpl) wRes.getParentAggregation()).getParentBean().getEuropeanaAggregation(), creatorMap);
+        // check for URI, labels and remove duplicate for creator
+        checkCreatorLabel(attribution, (List<Agent>) ((AggregationImpl) wRes.getParentAggregation()).getParentBean().getAgents(), creatorMap);
     }
 
-    private void checkProxy(List<? extends Proxy> proxies, Attribution attribution) {
+    private void checkProxy(List<? extends Proxy> proxies, Attribution attribution, Map<String, List<String>> creatorMap) {
         if (!proxies.isEmpty()) {
             for (Proxy proxy : proxies) {
-                if (attribution.getCreator().size() == 0 && isNotBlank(proxy.getDcCreator())) {
-                    attribution.setCreator(concatLangawareMap(proxy.getDcCreator()));
+                if (creatorMap.size() == 0 && isNotBlank(proxy.getDcCreator())) {
```

Comment:
you might consider using {Map}.isEmpty() instead of {Map}.size() == 0

Reply:
or maybe this was already in the code, not sure. I changed this in some other places further down